docs/proposals/rollout/rollout.md

```diff
@@ -386,12 +398,57 @@ type Metric struct {
     ThresholdRange *CanaryThresholdRange `json:"thresholdRange,omitempty"`
 
     // TemplateRef references a metric template object
+    // The metric can be custom by users.
     // +optional
-    TemplateRef *CrossNamespaceObjectReference `json:"templateRef,omitempty"`
+    TemplateRef *MetricTemplate `json:"templateRef,omitempty"`
+}
+
+// MetricTemplate defines custom metric template.
+// The results were obtained by querying Prometheus metrics and 
+// combining the calculations to validate the preview service.
+// Note: If use delete `Application.Spec.ApplicationSyncPolicy.Rollout` or
+// `Application`, Kurator will delete metricTemplate.
+type MetricTemplate struct {
+    // Name of MetricTemplate
+    // Name must be unique within a namespace.
+    // More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names#names
+    Name string `json:"name,omitempty"`
+
+    // Namespace of MetricTemplate
+    // More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces
+    Namespace string `json:"namespace,omitempty"`
+
+    // MetricProvider defines the metric provider of metricTemplate.
+    // Kurator now only supports `Prometheus` as metricProvider.
+    // Default is prometheus
+    MetricProvider string `json:"metricProvider,omitempty"`
```

Comment:
It is not clear how these fields work, i would suggest removing `MetricTemplate` ATM